src/scripts/main.js

```diff
@@ -1,11 +1,36 @@
+/* eslint-disable prettier/prettier */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const promise = new Promise((resolve) => {
+    element.addEventListener(eventName, (e) => {
+      e.preventDefault();
+
+      // eslint-disable-next-line no-undef
+      resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+    },
+    { once: true });
+  });
+
+  return promise;
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const existingMessage = document.querySelector('.message');
+
+  if (existingMessage) {
+    existingMessage.remove();
+  }
```

Comment:
This logic for removing an existing message before creating a new one can lead to a flickering effect, as the old message is removed and a new one is added. It could be better to simply update the text of the existing message if it exists.